utils.py

```diff
@@ -0,0 +1,56 @@
+import os
+import pandas as pd
+
+def df_info(df):
+    print('-------------------------------------------shape----------------------------------------------------------------')
+    print(df.shape)
+    print('-------------------------------------head() and tail(1)---------------------------------------------------------')
+    display(df.head(), df.tail(1))
+    print('------------------------------------------nunique()-------------------------------------------------------------')
+    print(df.nunique())
+    print('-------------------------------------describe().round()---------------------------------------------------------')
+    print(df.describe().round())
+    print('--------------------------------------------info()--------------------------------------------------------------')
+    print(df.info())
+    print('-------------------------------------------isnull()-------------------------------------------------------------')
+    print(df.isnull().sum())
+    print('--------------------------------------------isna()--------------------------------------------------------------')
+    print(df.isna().sum())
+    print('-----------------------------------------duplicated()-----------------------------------------------------------')
+    print(len(df[df.duplicated()]))
+    print('----------------------------------------------------------------------------------------------------------------') 
+
+def create_directory_structure():
```

Comment:
Why is this a function? Surely you just needed to do this once? Youcould have quicker made this structure manually and then added them to the repo so they were set up for others to use. Could imagine this being more useful if you had a folder for each run (perhaps with a date of runtime or dataset name in the title) and you want to generate it programmatically every time you run the optimisation.